ports/esp32s2/common-hal/pwmio/PWMOut.c

```diff
@@ -129,6 +129,12 @@ pwmout_result_t common_hal_pwmio_pwmout_construct(pwmio_pwmout_obj_t *self,
         return PWMOUT_INITIALIZATION_ERROR;
     }
 
+    // check the frequency -- 0 is not valid
+    // maybe also want to reject frequency > system_clock / 2 ??
+    if (frequency == 0) {
+        return PWMOUT_INVALID_FREQUENCY;
+    }
```

Comment:
```suggestion
    // check the frequency -- 0 is not valid
    if (frequency == 0) {
        return PWMOUT_INVALID_FREQUENCY;
    }
```

Please move this to line 66 as the crash seems to happen at line 68.

https://github.com/adafruit/circuitpython/blob/8613b9a9fef9c639fdc6eaf0fe4ec8c384c0aef3/ports/esp32s2/common-hal/pwmio/PWMOut.c#L66-L68

Reply:
Yeah, wow.